WiFi: winc1500: add new config flag

rdsingh-iot opened the pull request:
There is a hardcoded macro in Winc1500 HAL SPI driver

that would prevent the driver from working.

 This macro is now defined only based on KConfig entry.

 This KConfig entry is by default not set.

 To enable, use "CONFIG_WINC1500_DRV_USE_OLD_SPI_SW=y

in proj.conf or board.conf

Signed-off-by: Raja D.Singh <rdsingh@iotwizards.com>

jukkar commented on drivers/wifi/winc1500/Kconfig.winc1500:
```diff
@@ -15,6 +15,12 @@ config WIFI_WINC1500_NAME
 	string "Driver name"
```
Please change the commit subject to `drivers: wifi: winc1500: Add new config flag`
Also no need to put so many empty lines to the commit message.
So the commit message should look something like this:
```
drivers: wifi: winc1500: Add new config flag

There is a hardcoded macro in Winc1500 HAL SPI driver
that would prevent the driver from working.
This macro is now defined only based on KConfig entry.
This KConfig entry is by default not set.
To enable, set "CONFIG_WINC1500_DRV_USE_OLD_SW=y"
in proj.conf or board.conf

Signed-off-by: Raja D.Singh <rdsingh@iotwizards.com>
```

rdsingh-iot replied:
> Please change the commit subject to `drivers: wifi: winc1500: Add new config flag`
> Also no need to put so many empty lines to the commit message.
> So the commit message should look something like this:
> 
> ```
> drivers: wifi: winc1500: Add new config flag
> 
> There is a hardcoded macro in Winc1500 HAL SPI driver
> that would prevent the driver from working.
> This macro is now defined only based on KConfig entry.
> This KConfig entry is by default not set.
> To enable, set "CONFIG_WINC1500_DRV_USE_OLD_SW=y"
> in proj.conf or board.conf
> 
> Signed-off-by: Raja D.Singh <rdsingh@iotwizards.com>
> ```

OK.

rdsingh-iot replied:
Done.

jukkar commented on drivers/wifi/winc1500/Kconfig.winc1500:
```diff
@@ -15,6 +15,12 @@ config WIFI_WINC1500_NAME
 
+config WINC1500_DRV_USE_OLD_SPI_SW
+	bool "Driver to use Old SPI software"
+	default n
```
The option value is `n` by default, so this line can be omitted.

rdsingh-iot replied:
Sure.

rdsingh-iot replied:
Done.

jukkar reviewed (commented):
LGTM. The PR can be merged after the https://github.com/zephyrproject-rtos/hal_atmel/pull/12 is merged (you should replace the revision with the real commit id when that id is known).

rdsingh-iot commented:
> LGTM

@jukkar Thank you! 

Do I need to do anything for merging https://github.com/zephyrproject-rtos/hal_atmel/pull/12 ?

pfalcon commented on drivers/wifi/winc1500/Kconfig.winc1500:
```diff
@@ -15,6 +15,11 @@ config WIFI_WINC1500_NAME
 	default "WINC1500"
 
+config WINC1500_DRV_USE_OLD_SPI_SW
+	bool "Driver to use Old SPI software"
```
Please replace the description with "Use old SPI access method in the vendor HAL". And the help text: "This option, when enabled, causes USE_OLD_SPI_SW setting to be passed to vendor HAL".

rdsingh-iot replied:
@pfalcon Do you want the issue description to be - "Use old SPI access method in the vendor HAL"??

pfalcon replied:
> Do you want the issue description to be - "Use old SPI access method in the vendor HAL"??

No, that wouldn't properly describe the nature of the patch.  As you can fit only so much information into a single line of <70 chars, better let it stay like it is.

rdsingh-iot replied:
@pfalcon, I thought so too. Then what exactly do you mean by "Please replace the description with "Use old SPI access method in the vendor HAL".  Please exlain.

I understand what you wanted me to do with the help text.

pfalcon replied:
> Then what exactly do you mean by "Please replace the description with "Use old SPI access method in the vendor HAL". Please explain.

Currently the line in your patch to which this comment is attached reads:

~~~
	bool "Driver to use Old SPI software"
~~~

instead, it's suggested to be:
~~~
	bool "Use old SPI access method in the vendor HAL"
~~~

rdsingh-iot replied:
That really makes sense! Thanks.

rdsingh-iot replied:
OK, done. Please check. I will wait for the HAL pull request to go through, so that I can use the commit ID to make a final commit here after I squash them all in one commit.

pfalcon reviewed (changes requested):
Some suggestions to clarify Kconfig option descriptions.

rdsingh-iot commented:
> LGTM. The PR can be merged after the [zephyrproject-rtos/hal_atmel#12](https://github.com/zephyrproject-rtos/hal_atmel/pull/12) is merged (you should replace the revision with the real commit id when that id is known).

@jukkar Looks like someone needs to merge https://github.com/zephyrproject-rtos/hal_atmel/pull/12 so that this PR can be merged. Could you or someone else look in to it?

jukkar commented:
> Looks like someone needs to merge [zephyrproject-rtos/hal_atmel#12](https://github.com/zephyrproject-rtos/hal_atmel/pull/12) so that this PR can be merged.

@stephanosio are you maintaining atmel hal module?

rdsingh-iot commented:
> This pull request has been marked as stale because it has been open (more than) 60 days with no activity. Remove the stale label or add a comment saying that you would like to have the label removed otherwise this pull request will automatically be closed in 14 days. Note, that you can always re-open a closed pull request at any time.

@galak Hi, Wondering if you still need to see some changes to https://github.com/rdsingh-iot/hal_atmel/tree/remove_define upon which this PR is dependent.

jukkar commented:
This seems to be stalled because no activity on https://github.com/zephyrproject-rtos/hal_atmel/pull/12
@galak can you recheck https://github.com/zephyrproject-rtos/hal_atmel/pull/12 so we can unblock this one?

tbursztyka commented on drivers/wifi/winc1500/Kconfig.winc1500:
```diff
@@ -16,9 +16,10 @@ config WIFI_WINC1500_NAME
 
 config WINC1500_DRV_USE_OLD_SPI_SW
-	bool "Driver to use Old SPI software"
+	bool "Use old SPI access method in the vendor HAL"
```
Please merge this commit with the first one so it all goes into one commit.

rdsingh-iot replied:
Yep. Done!

rdsingh-iot replied:
I am not sure what's happening with the workflow pipeline... it seems stuck. Any ideas?

rdsingh-iot replied:
Hi,
The CI/CD pipeline did not error out. Instead, it is waiting on some non-existing tasks. Could someone tell me how to proceed. I think it is beyond my reach to get to this problem since I do not have access rights.

regards,
RDS

jukkar replied:
Can you rebase the PR against latest upstream and force push, there is now conflict with west.yml

rdsingh-iot replied:
> Can you rebase the PR against latest upstream and force push, there is now conflict with west.yml

Hi @jukkar,
Done! Please check.
regards,
RDS

galak reviewed (commented):
I approved and merged the hal_atmel side change for this.

Commit:
drivers: wifi: winc1500: Add new config flag

There is a hardcoded macro in Winc1500 HAL SPI driver
that would prevent the driver from working.
This macro is now defined only based on KConfig entry.
This KConfig entry is by default not set.
To enable, set "CONFIG_WINC1500_DRV_USE_OLD_SW=y"
in proj.conf or board.conf

Signed-off-by: Raja D. Singh <rdsingh@iotwizards.com>
